integration/spark/src/main/java/org/apache/spark/sql/secondaryindex/query/SecondaryIndexQueryResultProcessor.java

```diff
@@ -362,6 +367,9 @@ private void processResult(List<CarbonIterator<RowBatch>> detailQueryResultItera
    */
   private Object getData(Object[] data, int index, DataType dataType) {
     if (data == null || data.length == 0) {
+      if (DataTypeUtil.isPrimitiveColumn(dataType)) {
```

Comment:
please check the impact for Timestamp type for this change and add test case

Reply:
There is no impact now, added testcase.